libqtile/widget/mpris2widget.py

```diff
@@ -68,25 +75,38 @@ def __init__(self, **config):
         self.is_playing = False
         self.scroll_timer = None
         self.scroll_counter = None
-        self.dbus_loop = None
-
-    def _configure(self, qtile, bar):
-        base._TextBox._configure(self, qtile, bar)
-
-        # we don't need to reconnect all the dbus stuff if we already
-        # connected it.
-        if self.dbus_loop is not None:
+        self.count = 0
+        self.connect_count = 0
+
+    async def _config_async(self):
+        await self._connect_dbus()
+
+    async def _connect_dbus(self):
+        if self.connect_count < self.MAX_RETRIES:
+            self.connect_count += 1
+            await self._setup()
+        else:
+            logger.warning("Exceeded max retries. "
+                           "Could not subscribe to dbus signals.")
```

Comment:
Why do we need that reconnect logic when it wasn't here before?

Reply:
It was there before. https://github.com/qtile/qtile/pull/1994/commits/a039b2e3fc494a37c9480eaf6181c7d1453c341f#diff-8606537352d6a3e9599d52fbe08da22e73f39c568bcbff00b56cef16416333c0L97

Reply:
I meant in the python-dbus version of the code. :)

Reply:
Fair point. Will remove.